libs/article-parser/src/lib/parser/ebi-parser.ts

```diff
@@ -0,0 +1,43 @@
+import {
+  ParsedArticle,
+  Parser,
+  ParsedArticleParagraph,
+  EbiParserOptions,
+} from '@foodmedicine/interfaces';
+import * as cheerio from 'cheerio';
+
+/**
+ * A parser for https://www.ebi.ac.uk/europepmc/webservices/rest/
+ */
+export const EbiParser: Parser<ParsedArticle> = {
+  parserF: async (xml: string, opts?: EbiParserOptions) => {
+    if (
+      !opts ||
+      !opts.parsedArticleHead ||
+      !opts.impacted ||
+      !opts.recommendation
+    )
+      throw 'Please add in the parsed head through the options';
+    const $ = cheerio.load(xml);
+    const paragraphTexts: string[] = $('p')
+      .map((i, el) => $(el).text())
+      .get();
+    const paragraphsProms: Promise<
+      ParsedArticleParagraph
+    >[] = paragraphTexts.map((paragraphText) =>
+      opts.getCorrelationScore(
+        paragraphText,
+        opts.impacted,
+        opts.recommendation
+      )
+    );
+    const paragraphs = await Promise.all(paragraphsProms);
+    const article: ParsedArticle = {
+      head: opts.parsedArticleHead,
+      paragraphs,
+    };
+    return [article];
```

Comment:
why return article in an array?

Reply:
It is in an array to match the spec of the Scraper class. I agree that it is not clean/ self explanatory, is there a better way to write this?

Reply:
For which part of the scraper class? another comment you had left suggested that the scraper took a ...url param specifically in order to allow for single URLs. Can you implement the same approach here?

Reply:
Oh ok, I added `IRet | IRet[]`

Reply:
In that case, should this just return `article` without the array now?

Reply:
Oh yep it does